Dockerfile

```diff
@@ -8,19 +8,20 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG BUILDER=golang:1.19-alpine
+ARG BUILDER=public.ecr.aws/eks-distro-build-tooling/golang:1.19-gcc
 ARG BASE_IMAGE=public.ecr.aws/eks-distro/kubernetes/go-runner:v0.13.0-eks-1-23-9
 
-FROM ${BUILDER} AS build
+FROM --platform=$BUILDPLATFORM ${BUILDER} AS build
 WORKDIR /go/src/sigs.k8s.io/aws-encryption-provider
 ARG TAG
 COPY . ./
 ENV GO111MODULE=on
-RUN	CGO_ENABLED=0 GOOS=linux go build -mod mod -ldflags \
+ARG TARGETOS TARGETARCH
+RUN CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH go build -mod mod -ldflags \
```

Comment:
Nit: I know this is carry over from before, but `-mod mod` is just the default right?  Should we leave it out because I've never seen it specified before.